### ISSUE TITLE
Support code based configuration and consider removing deps on the crusty .NET config system

### ISSUE THREAD

**Issue by Iristyle** [basho/riak-dotnet-client#4]
I have read this (and your announcement about intending to keep Xml), so am I familiar with your stance ;0
http://news.ycombinator.com/item?id=2799823

We are using your client internally against our test Riak cluster... and the config was one thing that immediately jumped out at me as something I didn't like.  Having done a lot of ASP.NET / MVC / WCF development, I have a very strong bias against XML config at all costs -- it tends to grow out of hand and become a real beast to read, understand and maintain over time.   

I would advocate for supporting both .config file based and code-based config, and based on your code, you can do this w/out a lot of extra effort.  You're already most of the way there given your config is expressed through interfaces already.  We're working around this by implementing the config interfaces ourselves with POCOs that have no connection to the .NET config classes, but I'd like to see you support that out of the box.  All you need to provide really are hand-rolled validation routines that match up with the config validation attributes.  (I've done this sort of thing internally on other projects)

I would also like to make a couple of comments about your remarks on the ycombinator thread.
- If you're going to setup a Riak cluster, which has to run on some flavor of _nix, I don't see why anyone would *not_ setup a load balancer like HAProxy.  I am decidedly _not_ a Debian expert, but it took me about 20 mins to setup HAproxy to load balance the cluster.  My point is, presumably any prod cluster will be setup by ops guys (or smart devs!) and setting up HAproxy is not hard... it would be one thing if Riak could run on Windows, but since it can't... I don't understand the stance there.
- By implementing load balancing in your client, you're more or less making the assumption that its the only one communicating with the cluster.  If there are other client stacks talking to the cluster, then the accuracy of your load balancing is diminished.   IMHO, it really shouldn't be the responsibility of the client toolkit to determine a load balancing strategy given it has no idea of what other clients might be doing.
- Recommendation for using protobufs is to use the 'least connected' algorithm since the protobuf connection is long running, which HAProxy and other proxies support out of the box.  Round robin isn't the best choice there.  To my previous point, this would be very difficult to implement correctly in CorrugatedIron if there are other clients communicating with Riak.  

There's a good chance that we'll be building out a few pieces on node.js for instance.  I think if you're a team looking at Riak, chances are you'll probably have at least a couple of client stacks working off it, which would again emphasize the need for a centralized load balancer.

So in short, I think it's awesome that your client does round robin load balancing -- and there are some scenarios where that might be useful.  However, I think in any real production use, I can't see a lot of people using that.  If I were to guess, I would expect that the majority of .NET teams would be deploying on Mono, where installing HAProxy is something that's more familiar anyhow.  In our world, we'll be building out a service layer with ServiceStack, and almost certainly be deploying on Mono.
